app/repositories/slick/implementations/SlickAuthenticationRepository.scala

```diff
@@ -129,13 +137,17 @@ class SlickAuthenticationRepository @Inject() (db: Database)(implicit executionC
   def updateToken(address: String): Future[Option[String]] =
     db.run(updateTokenAction(address).transactionally)
 
-  def getTokenExpirationDate(token: String): Future[Option[Timestamp]] =
-    db.run(TokensTable.all.filter(_.token === token).map(_.endDate).result.headOption)
-
-  def getUser(token: String): Future[Option[String]] =
-    db.run((for {
-      tokenId <- TokensTable.all.filter(_.token === token).map(_.tokenId)
-      userId <- PasswordsTable.all.filter(_.tokenId === tokenId).map(_.userId)
-    } yield userId).result.headOption)
+  def getUser(token: String): Future[Either[Error, String]] =
+    db.run(TokensTable.all.filter(_.token === token).exists.result).map(
+      if (_) {
+        JwtJson.decodeJson(token, key, Seq(algo)) match {
+          case Success(json) => json.validate[Jwt].fold(
+            errors => Left(internalError),
+            jwt => if (jwt.expirationDate.isAfter(LocalDateTime.now))
+              Right(jwt.userId)
+            else Left(tokenNotValid))
+          case Failure(e) => Left(internalError)
+        }
+      } else Left(tokenNotValid))
```

Comment:
You should check expiration before going to the db. That was the purpose of putting the date in the token.